### PR TITLE
feat: add custom logger that includes the origin

### PR DESCRIPTION
**Pull request by yrobla**
When running servers, use a custom structlog logger that binds an extra parameter called origin. Can be: generic_server, copilot_proxy

Closes: #301

### CODE REVIEW

**Comment by yrobla** [comment]
sample output:

image

File: src/codegate/logger/logger.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import structlog
+
+
+class OriginLogger:
+    def __init__(self, origin: str):
+        self.logger = structlog.get_logger().bind(origin=origin)
```
Comment (aponcedeleonch) [inline src/codegate/logger/logger.py:6]: I think this needs to be
```
self.logger = structlog.get_logger("codegate").bind(origin=origin)
```

Or at least with this change I'm getting `debug` logs even though I set in the cli the logging level to INFO
```
2024-12-18 12:33:09 [debug    ] Processing body through pipeline: 0 bytes origin=copilot_proxy
2024-12-18 12:33:09 [debug    ] No pipeline strategy selected  origin=copilot_proxy
2024-12-18 12:33:09 [debug    ] Extracting path from /extensions/chat.json origin=copilot_proxy
2024-12-18 12:33:09 [debug    ] Processing body through pipeline: 0 bytes origin=copilot_proxy
2024-12-18 12:33:09 [debug    ] No pipeline strategy selected  origin=copilot_proxy
```

File: src/codegate/providers/copilot/provider.py [hunk 2]
```diff
@@ -22,7 +23,8 @@
 )
 from codegate.providers.copilot.streaming import SSEProcessor
 
-logger = structlog.get_logger("codegate")
+setup_logging()
```
Comment (aponcedeleonch) [inline src/codegate/providers/copilot/provider.py:26]: Do we need this call to `setup_logging()`, I think it might reconfigure the logger  on-the-fly?

**Review by aponcedeleonch — APPROVED** [review]
Nothing, just tested works nicely!

File: src/codegate/providers/copilot/provider.py (hunk 2 above)
Comment (yrobla) [reply src/codegate/providers/copilot/provider.py:26]: if we do not set it, copilot loses the scope and is not logging with the expected format

**Pull request merged**